models/Asset/Listing.php

```diff
@@ -79,19 +79,20 @@ public function getItems($offset, $itemCountPerPage)
      * @internal
      *
      * @param Model\User $user
+     * @param string $permission
      *
      * @return static
      */
-    public function filterAccessibleByUser(Model\User $user)
+    public function filterAccessibleByUser(Model\User $user, $permission = 'list')
     {
         if (!$user->isAdmin()) {
             $userIds = $user->getRoles();
             $userIds[] = $user->getId();
 
             $condition = '(
-                (SELECT list FROM users_workspaces_asset WHERE userId IN ('.implode(',', $userIds).') AND LOCATE(CONCAT(path,filename),cpath)=1 ORDER BY LENGTH(cpath) DESC, FIELD(userId, '.$user->getId().') DESC, list DESC LIMIT 1)=1
+                (SELECT `'.$permission.'` FROM users_workspaces_asset WHERE userId IN ('.implode(',', $userIds).') AND LOCATE(CONCAT(path,filename),cpath)=1 ORDER BY LENGTH(cpath) DESC, FIELD(userId, '.$user->getId().') DESC, `'.$permission.'` DESC LIMIT 1)=1
```

Comment:
why not use `$this->quote($permission)` here?

Reply:
`$this->quote()` adds quotes(`'`) but these are database column names, here we need backticks (`)

Reply:
I guess $db->quoteIdentifier() would do the trick.

Reply:
Ok, I changed it to `Db::get()->quoteIdentifier($permission)`.
Actually I cannot understand such pickiness while in other places it is implemented exactly how I had it with the backticks, see for example https://github.com/pimcore/pimcore/blob/86ab63e336ce5d36201370381b3aaf177e56b435/models/Asset/Dao.php#L430

Reply:
Thank you very much and sorry for that pickiness ... but better safe than sorry. 
The simple backticks approach opens up the door for SQL injection. Just use ```$type = 'list` FROM assets; DELETE assets; -- ';``` as `$type` and bang. And when we already touch things, we should do them right. 

Besides that, we also need to fix the other places too of course ... see https://github.com/pimcore/pimcore/issues/9412